### PR TITLE
add get_dims() API

### PR DESCRIPTION
**Pull request by okdshin**
This PR adds `menoh_variable_profile_table_get_dims()` and `menoh_model_get_variable_dims()`.
See #48

### CODE REVIEW

File: example/vgg16_example_in_c.c [hunk 1]
```diff
@@ -42,11 +44,22 @@ int main() {
...
       variable_profile_table, softmax_out_name, 1, &softmax_out_dims[1]));
+    */
+
+    int32_t softmax_out_dims_size;
+    const int32_t* softmax_out_dims;
+    ERROR_CHECK(menoh_variable_profile_table_get_dims(
+      variable_profile_table, softmax_out_name, &softmax_out_dims_size,
+      &softmax_out_dims));
+    assert(softmax_out_dims_size == 1);
+    assert(softmax_out_dims[0]== 2);
```
Comment (msakai) [inline example/vgg16_example_in_c.c]: I think this should be:
```
assert(softmax_out_dims_size == 2);
assert(softmax_out_dims[0]== 1);
```

File: example/vgg16_example_in_c.c [hunk 2]
```diff
@@ -42,11 +44,22 @@ int main() {
     ERROR_CHECK(menoh_build_variable_profile_table(vpt_builder, model_data,
                                                    &variable_profile_table));
 
+    /*
     int32_t softmax_out_dims[2];
     ERROR_CHECK(menoh_variable_profile_table_get_dims_at(
       variable_profile_table, softmax_out_name, 0, &softmax_out_dims[0]));
     ERROR_CHECK(menoh_variable_profile_table_get_dims_at(
       variable_profile_table, softmax_out_name, 1, &softmax_out_dims[1]));
+    */
```
Comment (msakai) [inline example/vgg16_example_in_c.c]: Is it necessary to keep the old code as comment?

File: example/vgg16_example_in_c.c (hunk 1 above)
Comment (okdshin) [reply example/vgg16_example_in_c.c]: Oops, I will modify it. Thanks.

File: example/vgg16_example_in_c.c (hunk 2 above)
Comment (okdshin) [reply example/vgg16_example_in_c.c]: I won't deprecate `get_dims_size()` and `get_dims_at()`. They are usable in some situations so I would like to show their usage as well. Do you have any idea?

Comment (msakai) [reply example/vgg16_example_in_c.c]: I see. That makes sense.

**Comment by okdshin** [comment]
Thank you for reviwing!

**Pull request merged**